src/main/plugin/iso19139.ca.HNAP/layout/utility-tpl-multilingual.xsl

```diff
@@ -112,7 +137,51 @@
 
   <!-- Get the list of other languages -->
   <xsl:template name="get-iso19139.ca.HNAP-other-languages">
-    <xsl:call-template name="get-iso19139-other-languages" />
+    <xsl:variable name="isTemplate" select="$metadata/gn:info[position() = last()]/isTemplate"/>
+    <xsl:choose>
+      <xsl:when test="$isTemplate = 's' or $isTemplate = 't'">
+
+        <xsl:for-each select="distinct-values($metadata//gmd:LocalisedCharacterString/@locale)">
+          <xsl:variable name="locale" select="string(.)"/>
+          <xsl:variable name="langId" select="xslutil:threeCharLangCode(substring($locale,2,2))"/>
+          <lang id="{substring($locale,2,2)}" code="{$langId}"/>
+        </xsl:for-each>
+      </xsl:when>
+      <xsl:otherwise>
+        <xsl:variable name="mainLanguage">
+          <xsl:call-template name="get-iso19139-language"/>
+        </xsl:variable>
+        <xsl:for-each select="$metadata/gmd:locale/gmd:PT_Locale">
+          <xsl:variable name="langCode"
+                        select="gmd:languageCode/gmd:LanguageCode/@codeListValue"/>
+          <xsl:variable name="id_ISO639_2B">
+            <xsl:choose>
+              <xsl:when test="@id='fra'">
```

Comment:
Should not be comparing to the @id as it is a variable name?

Reply:
I should not have this variable at all. The id should be passed as lang/@id as is. The function is copied pasted from ISO19139. I did minor change by converting language code here.

Reply:
So why are you converting an id from fra to fre.  If the id in the xml is called `fra` then it will be referenced as `#fra` so changine the variable to fre seems incorrect.  If you want fre then it seems like it is using langCode_ISO639_2B. I'm not sure I follow the logic of changing variable names in the code?

Reply:
This was copied from ISO19139 profile and I attempt to convert both of its id and value to fre:#fre. The key:value will be passed to other module as I understand. I don't the full context of that id, but if the id is passed as fra, then we will have issue in English mode when the other language (French) cannot be detected.